builtin/logical/aws/path_user.go

```diff
@@ -142,11 +142,11 @@ func (b *backend) pathCredsRead(ctx context.Context, req *logical.Request, d *fr
 	case assumedRoleCred:
 		switch {
 		case roleArn == "":
-			if len(role.RoleArns) != 1 {
+			if len(role.RoleArns) != 1 || strings.Contains(role.RoleArns[0], "*") {
```

Comment:
I'm not sure that this would work, as the engine would still need to know a specific role (ARN) to assume.

Reply:
I think the appropriate check should be the following, where we only set `roleArn = role.RoleArns[0]`, if the role we're trying to infer is not a glob. The glob-only case should still be covered by second case in this switch statement.

```suggestion
			if len(role.RoleArns) != 1 && !strings.Contains(role.RoleArns[0], "*") {
```

Reply:
I totally misread this, and this we're covering an error case here! Your check is correct.